baker/SiteBaker.tsx

```diff
@@ -431,8 +431,6 @@ export class SiteBaker {
     private async removeDeletedPosts(knex: db.KnexReadonlyTransaction) {
         if (!this.bakeSteps.has("removeDeletedPosts")) return
 
-        await db.getConnection()
```

Comment:
I don't think this change has come from here, but I also checked in the head of this PR chain and it's happening there too so I'll just write it here that:

```
node itsJustJavascript/baker/buildLocalBake.js --steps specialPages
```
with
```
TypeError: Cannot read properties of null (reading 'toLocaleDateString')
    at formatDate (grapher/packages/@ourworldindata/utils/dist/Util.js:960:17)
    at grapher/itsJustJavascript/db/model/Gdoc/GdocBase.js:674:49
    at Array.map (<anonymous>)
    at getMinimalGdocPostsByIds (grapher/itsJustJavascript/db/model/Gdoc/GdocBase.js:668:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async GdocHomepage.loadLinkedDocuments (grapher/itsJustJavascript/db/model/Gdoc/GdocBase.js:507:33)
    at async GdocHomepage.loadState (grapher/itsJustJavascript/db/model/Gdoc/GdocBase.js:626:9)
    at async loadGdocFromGdocBase (grapher/itsJustJavascript/db/model/Gdoc/GdocFactory.js:199:5)
    at async renderFrontPage (grapher/itsJustJavascript/baker/siteRenderers.js:172:30)
    at async SiteBaker.bakeSpecialPages (grapher/itsJustJavascript/baker/SiteBaker.js:353:66)
```